### ISSUE TITLE
Openstack provider: Bad network format: missing 'uuid'

### ISSUE THREAD

**Issue by james147** [hashicorp/terraform#2031]
Terraform version: v0.5.2

I get the error `Bad network format: missing 'uuid'` when running `terraform apply` with the following tf file:

```
# Configure the OpenStack Provider
provider "openstack" {
  # Uses env vars
}

# Create a web server
resource "openstack_compute_instance_v2" "test-server" {
  name = "tf-test-1"
  region = "RegionOne"
  image_id = "ee026709-df4c-4559-8613-c7fafd85a30c"
  flavor_id = "8de01de0-51c4-458f-baba-39fdaeff7ceb"
  network {
    uuid = "3522b3e0-f729-490d-b9e3-9ffd025719df"
  }
}
```

It looks like the uuid is being picked up from the config, but not passed to the provider. Running `TF_LOG=1 terraform apply` produces the following output:

```
...
openstack_compute_instance_v2.test-server: Creating...
  access_ip_v4:          "" => "<computed>"
  access_ip_v6:          "" => "<computed>"
It looks like the uuid is being picked up from the config, but not passed to the provider.
  flavor_id:             "" => "8de01de0-51c4-458f-baba-39fdaeff7ceb"
  flavor_name:           "" => "<computed>"
  image_id:              "" => "ee026709-df4c-4559-8613-c7fafd85a30c"
  image_name:            "" => "<computed>"
  name:                  "" => "tf-test-1"
  network.#:             "" => "1"
  network.0.fixed_ip_v4: "" => "<computed>"
  network.0.fixed_ip_v6: "" => "<computed>"
  network.0.mac:         "" => "<computed>"
  network.0.name:        "" => "<computed>"
  network.0.port:        "" => "<computed>"
2015/05/21 11:16:11 [DEBUG] apply: openstack_compute_instance_v2.test-server: executing Apply
  network.0.uuid:        "" => "3522b3e0-f729-490d-b9e3-9ffd025719df"
  region:                "" => "RegionOne"
2015/05/21 11:16:12 terraform-provider-openstack: 2015/05/21 11:16:12 [DEBUG] networks: [map[uuid: name: port: fixed_ip_v4:]]
2015/05/21 11:16:12 terraform-provider-openstack: 2015/05/21 11:16:12 [DEBUG] Create Options: &servers.CreateOpts{Name:"tf-test-1", ImageRef:"ee026709-df4c-4559-8613-c7fafd85a30c", FlavorRef:"8de01de0-51c4-458f-baba-39fdaeff7ceb", SecurityGroups:[]string{}, UserData:[]uint8{}, AvailabilityZone:"", Networks:[]servers.Network{servers.Network{UUID:"", Port:"", FixedIP:""}}, Metadata:map[string]string{}, Personality:servers.Personality(nil), ConfigDrive:false, AdminPass:"", AccessIPv4:"", AccessIPv6:""}
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalWriteState
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalIf
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalWriteDiff
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalIf
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalWriteState
2015/05/21 11:16:12 [DEBUG] root: eval: *terraform.EvalApplyPost
2015/05/21 11:16:12 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}
2015/05/21 11:16:12 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}
2015/05/21 11:16:12 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}
2015/05/21 11:16:12 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}
openstack_compute_instance_v2.test-server: Error: 1 error(s) occurred:

2015/05/21 11:16:12 waiting for all plugin processes to complete...
* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}
Error applying plan:

1 error(s) occurred:

* 1 error(s) occurred:

* 1 error(s) occurred:

* Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST http://.../v2/f7a779e6bda64d3db9cfc9cae34ed931/servers], but got 400 instead
{"badRequest": {"message": "Bad network format: missing 'uuid'", "code": 400}}

Terraform does not automatically rollback in the face of errors.
2015/05/21 11:16:12 [DEBUG] /usr/local/bin/terraform-provider-openstack: plugin process exited
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

If the network group is left out the vm is booted with all existing networking interfaces attached to it. If the network block is then added `TF_LOG=1 terraform apply` produces the following output and the vms network interfaces do not change:

```
openstack_compute_instance_v2.test-server: Modifying...
  network.0.uuid:    "" => "3522b3e0-f729-490d-b9e3-9ffd025719df"
  security_groups.#: "1" => "0"
  security_groups.0: "default" => ""
2015/05/21 11:52:56 [DEBUG] apply: openstack_compute_instance_v2.test-server: executing Apply
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] Security groups to add: &{0x481f80 map[] {{0 0} 1}}
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] Security groups to remove: &{0x481f10 map[3814588639:default] {{0 0} 1}}
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] Removed security group (default) from instance (7ebc0815-ff6a-4bee-b727-b381afdc658e)
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] Retreived Server 7ebc0815-ff6a-4bee-b727-b381afdc658e: &{ID:7ebc0815-ff6a-4bee-b727-b381afdc658e TenantID:f7a779e6bda64d3db9cfc9cae34ed931 UserID:mdaffin Name:tf-test-1 Updated:2015-05-21T11:52:28Z Created:2015-05-21T11:51:35Z HostID:aa48f0e68950275a74e4215c06c8e03f94c15eacaf909e0710709d1b Status:ACTIVE Progress:0 AccessIPv4: AccessIPv6: Image:map[links:[map[href:http://10.3.0.2:8774/f7a779e6bda64d3db9cfc9cae34ed931/images/ee026709-df4c-4559-8613-c7fafd85a30c rel:bookmark]] id:ee026709-df4c-4559-8613-c7fafd85a30c] Flavor:map[id:8de01de0-51c4-458f-baba-39fdaeff7ceb links:[map[href:http://10.3.0.2:8774/f7a779e6bda64d3db9cfc9cae34ed931/flavors/8de01de0-51c4-458f-baba-39fdaeff7ceb rel:bookmark]]] Addresses:map[Adserver5:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:87:13:8d version:4 addr:10.3.35.6 OS-EXT-IPS:type:fixed]] BigData2:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:bd:d7:9e version:4 addr:10.3.17.2 OS-EXT-IPS:type:fixed]] BigData3:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:c9:ed:b3 version:4 addr:10.3.18.2 OS-EXT-IPS:type:fixed]] BigData1:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:73:34:fa version:4 addr:10.3.16.2 OS-EXT-IPS:type:fixed]] MySwads:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:23:8b:fb version:4 addr:10.3.8.2 OS-EXT-IPS:type:fixed]] Quill:[map[version:4 addr:10.3.22.3 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:59:c7:2c]] Infrastructure:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:e9:8e:77 version:4 addr:10.3.20.18 OS-EXT-IPS:type:fixed]] Trubshaw:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:db:8e:1c version:4 addr:10.3.23.2 OS-EXT-IPS:type:fixed]] Adserver2:[map[addr:10.3.32.6 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:68:c1:b4 version:4]] Adserver1:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:f8:3a:37 version:4 addr:10.3.31.2 OS-EXT-IPS:type:fixed]] Developer:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:e8:eb:6c version:4 addr:10.3.21.4 OS-EXT-IPS:type:fixed]] Adserver3:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:8b:c0:f2 version:4 addr:10.3.33.2 OS-EXT-IPS:type:fixed]] Adserver4:[map[OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:fb:ea:0b version:4 addr:10.3.34.6]] TBN-WAN:[map[version:4 addr:149.7.58.20 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:b6:ca:9a]]] Metadata:map[] Links:[map[href:http://10.3.0.2:8774/v2/f7a779e6bda64d3db9cfc9cae34ed931/servers/7ebc0815-ff6a-4bee-b727-b381afdc658e rel:self] map[href:http://10.3.0.2:8774/f7a779e6bda64d3db9cfc9cae34ed931/servers/7ebc0815-ff6a-4bee-b727-b381afdc658e rel:bookmark]] KeyName: AdminPass: SecurityGroups:[]}
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] networks: [map[uuid: name: port: fixed_ip_v4:]]
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] Addresses: map[Adserver1:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:f8:3a:37 version:4 addr:10.3.31.2 OS-EXT-IPS:type:fixed]] Developer:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:e8:eb:6c version:4 addr:10.3.21.4 OS-EXT-IPS:type:fixed]] Adserver4:[map[version:4 addr:10.3.34.6 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:fb:ea:0b]] TBN-WAN:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:b6:ca:9a version:4 addr:149.7.58.20 OS-EXT-IPS:type:fixed]] Adserver3:[map[addr:10.3.33.2 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:8b:c0:f2 version:4]] Adserver5:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:87:13:8d version:4 addr:10.3.35.6 OS-EXT-IPS:type:fixed]] Infrastructure:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:e9:8e:77 version:4 addr:10.3.20.18 OS-EXT-IPS:type:fixed]] Trubshaw:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:db:8e:1c version:4 addr:10.3.23.2 OS-EXT-IPS:type:fixed]] Adserver2:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:68:c1:b4 version:4 addr:10.3.32.6 OS-EXT-IPS:type:fixed]] BigData2:[map[OS-EXT-IPS-MAC:mac_addr:fa:16:3e:bd:d7:9e version:4 addr:10.3.17.2 OS-EXT-IPS:type:fixed]] BigData3:[map[OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:c9:ed:b3 version:4 addr:10.3.18.2]] BigData1:[map[version:4 addr:10.3.16.2 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:73:34:fa]] MySwads:[map[addr:10.3.8.2 OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:23:8b:fb version:4]] Quill:[map[OS-EXT-IPS:type:fixed OS-EXT-IPS-MAC:mac_addr:fa:16:3e:59:c7:2c version:4 addr:10.3.22.3]]]
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 [DEBUG] new networks: [map[fixed_ip_v6:<nil> mac:<nil> uuid: name: port: fixed_ip_v4:<nil>]]
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 hostv4:
2015/05/21 11:52:56 terraform-provider-openstack: 2015/05/21 11:52:56 hostv6:
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalWriteState
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalIf
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalWriteDiff
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalIf
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalWriteState
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalApplyPost
2015/05/21 11:52:57 [DEBUG] root: eval: *terraform.EvalUpdateStateHook
openstack_compute_instance_v2.test-server: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
2015/05/21 11:52:57 waiting for all plugin processes to complete...
2015/05/21 11:52:57 [DEBUG] /usr/local/bin/terraform-provider-openstack: plugin process exited
```

**Comment by jtopjian** [comment]
Hi there,

Are you still seeing this problem? I'm unable to reproduce it.

If you are still seeing it, can you confirm that the tenant/project you're using has access (is able to launch) instances on that network?

Thanks,
Joe

**Comment by james147** [comment]
I will check again tomorrow. But the tenant was able to launch instances with that network and without specifying a network terraform launched an instance with all networks attached that we had defined.

**Comment by jtopjian** [comment]
Sounds good :+1:

**Comment by james147** [comment]
I can no longer replicate this issue with the latest version. Thanks for looking into it :)

**Comment by bartekrutkowski** [comment]
What was the version that had this bug and the version that had it fixed? I am now using 0.6.9 (from homebrew) and I am experiencing exactly the same issue, uuid variable is not being populated:

> 2016/01/28 10:08:28 [DEBUG] terraform-provider-openstack: 2016/01/28 10:08:28 [DEBUG] Create Options: &servers.CreateOpts{Name:"vm-01.example.com", ImageRef:"a2361803-4c4b-4d64-91a2-d621da4662e6", ImageName:"", FlavorRef:"4", FlavorName:"", SecurityGroups:[]string{}, UserData:[]uint8{}, AvailabilityZone:"", Networks:[]servers.Network{servers.Network{UUID:"", Port:"", FixedIP:"192.168.10.11"}}, Metadata:map[string]string{}, Personality:servers.Personality(nil), ConfigDrive:false, AdminPass:"", AccessIPv4:"", AccessIPv6:""}
